### ISSUE TITLE
Failure to Parse Host Names with Numbers

### ISSUE THREAD

**Issue by fgauthier** [versatica/JsSIP#196]
A configuration with 'ws_servers' set to something like 'ws://myhost123:4568/somepath' is incorrectly flagged as illegal due to the '123' chars in the hostname.

The problem occurs in the 'parse_toplabel()' function where /^[a-zA-Z_\-]/' is used throughout as a matching pattern although this constraint should only apply to the first character of the host name. Using '/^[a-zA-Z0-9_\-]/' instead (after the first char) fixes this issue.

In short: the first character must be a letter, middle characters can be letters, numbers or hyphens and the last char can only be a letter or a number. See: http://www.ietf.org/rfc/rfc1034.txt.

**Comment by jmillan** [comment]
Finally,

'domainlabel' and 'toplable' rules have been relaxed from RFC3261 by allowing these to terminate with dash or hypen. They are not widely relaxed as before (allowing any string that contains [A-Za-z0-9._-]+): they must start with 'alphanum' and 'ALPHA' respectively as stated by the spec.

I've included the tests you made @ibc. All test pass.

**Comment by ibc** [comment]
:+1:
